pkg/cmd/adm/restart.go

```diff
@@ -1,157 +1,16 @@
-	return nil
-}
-
-func restartHostOperator(ctx *clicontext.CommandContext, hostClient runtimeclient.Client, hostNamespace string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := hostClient.List(context.TODO(), deployments,
-		runtimeclient.InNamespace(hostNamespace),
-		runtimeclient.MatchingLabels{"olm.owner.namespace": "toolchain-host-operator"}); err != nil {
-		return err
-	}
-	if len(deployments.Items) != 1 {
-		return fmt.Errorf("there should be a single deployment matching the label olm.owner.namespace=toolchain-host-operator in %s ns, but %d was found. "+
-			"It's not possible to restart the Host Operator deployment", hostNamespace, len(deployments.Items))
-	}
-
-	return restartDeployment(ctx, hostClient, hostNamespace, deployments.Items[0].Name)
-}
-
-func printExistingDeployments(term ioutils.Terminal, cl runtimeclient.Client, ns string) error {
-	deployments := &appsv1.DeploymentList{}
-	if err := cl.List(context.TODO(), deployments, runtimeclient.InNamespace(ns)); err != nil {
-		return err
-	}
-	deploymentList := "\n"
-	for _, deployment := range deployments.Items {
-		deploymentList += fmt.Sprintf("%s\n", deployment.Name)
-	}
-	term.PrintContextSeparatorWithBodyf(deploymentList, "Existing deployments in %s namespace", ns)
-	return nil
-}
-
-func scaleToZero(cl runtimeclient.Client, namespacedName types.NamespacedName) (int32, error) {
-	// get the deployment
-	deployment := &appsv1.Deployment{}
-	if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-		return 0, err
-	}
-	// keep original number of replicas so we can bring it back
-	originalReplicas := *deployment.Spec.Replicas
-	zero := int32(0)
-	deployment.Spec.Replicas = &zero
-
-	// update the deployment so it scales to zero
-	return originalReplicas, cl.Update(context.TODO(), deployment)
-}
-
-func scaleBack(term ioutils.Terminal, cl runtimeclient.Client, namespacedName types.NamespacedName, originalReplicas int32) error {
-	return wait.Poll(500*time.Millisecond, 10*time.Second, func() (done bool, err error) {
-		term.Println("")
-		term.Printlnf("Trying to scale the deployment back to '%d'", originalReplicas)
-		// get the updated
-		deployment := &appsv1.Deployment{}
-		if err := cl.Get(context.TODO(), namespacedName, deployment); err != nil {
-			return false, err
-		}
-		// check if the replicas number wasn't already reset by a controller
-		if *deployment.Spec.Replicas == originalReplicas {
-			return true, nil
-		}
-		// set the original
-		deployment.Spec.Replicas = &originalReplicas
-		// and update to scale back
-		if err := cl.Update(context.TODO(), deployment); err != nil {
-			term.Printlnf("error updating Deployment '%s': %s. Will retry again...", namespacedName.Name, err.Error())
-			return false, nil
-		}
-		return true, nil
+	return kubectl.SetUpKubectlCmd(func(factory cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+		return kubectlrollout.NewCmdRolloutRestart(factory, ioStreams)
```

Comment:
as I mentioned in our call and also in the jira story, the command should also wait until the rollout is complete